doc/adr/0016-end-to-end-testing-strategy.rst

```diff
@@ -0,0 +1,277 @@
+.. _end_to_end_testing_strategy:
+
+ADR 16: End-to-end testing strategy for Plutus, cardano-ledger-api and cardano-node-client
+==========================================================================================
+
+Date: 2022-12-02
+
+Author(s)
+---------
+
+james-iohk <james.browning@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+End-to-end testing of `Plutus Core <https://github.com/input-output-hk/Plutus/>`_ functionality is
+currently performed by a combination of automation and exploratory approaches, both are performed
+on public `preview` and `pre-prod` testnets using a real node.
+Automation test scenarios for Plutus are currently being run as part of the wider `cardano-node-tests
+<https://github.com/input-output-hk/cardano-node-tests/>`_ test suite, which uses a Python wrapper
+for ``cardano-cli``.
+Those tests focus on general ledger/node/cli functionality and only cover a few key scenarios for
+Plutus functionality, such as TxInfo and SECP256k1
+builtins.
+
+There is also ongoing development work to separate the functionality of `cardano-api
+<https://github.com/input-output-hk/cardano-node/tree/master/cardano-api>`_ out into two packages:
+
+* `cardano-ledger-api
+  <https://github.com/input-output-hk/cardano-ledger/tree/master/libs/cardano-ledger-api>`_ handles
```

Comment:
I thought we are going to move `cardano-api` into `plutus-apps`. But then it will be split into two packages, neither of which lives in `plutus-apps`? What's the actual plan?

Reply:
I don't think there was a plan to move it to `plutus-apps`. The most probable would have been a separate repository, but in the meantime, it is going to stay in `cardano-node`.

I made the proposal to split cardano-api here: https://github.com/input-output-hk/cardano-node/discussions/4687

The second phase which I'll officially propose later is to move functions from `cardano-api` to `cardano-ledger-api`. Then `cardano-ledger-api` will be the official package to build transactions.